### ISSUE TITLE
SetupHelper v6.13 release approaching

### ISSUE THREAD

**Issue by kwindrem** [Lucifer06/RemoteGPIO#19]
I am getting ready to release SetupHelper v6.13 to the wild.

I just pushed v6.13~8 to the beta branch if you want to look at the latest changes.

Anything else I need to address first ?

(Also posted this issue in SetupHelper repo but wanted to make sure you were aware of it.)

**Comment by Lucifer06** [comment]
I have tested v6.13~8 on a fresh installation.

First was install from ssh of v6.10 then it asked for GUI restart.
Then from GUI I switched to beta branch and installed v6.13~8.
It didn't asked for a GUI restart, it went straight away with the GUI restart.

Then I requested the install of RemoteGPIO. Conflicts showed that GuiMods was missing, and not in the active list.
I just added GuiMods (no install) and went back to the conflict resolution.
I thought it would have downloaded and installed it automatically, but it  didn't happen.
[
![Capture d’écran 2024-03-14 à 16 46 05](https://github.com/Lucifer06/RemoteGPIO/assets/10178879/1827e063-3061-4d4d-9512-233fb82ef30e)
](url)

From there I was in kind of dead loop: 
If I go to GuiMods page I get the following page (same if I go to RemoteGPIO): it shows SetupHelper page, not the correct page:
![Capture d’écran 2024-03-14 à 16 49 10](https://github.com/Lucifer06/RemoteGPIO/assets/10178879/8fbcd604-69c7-4012-99d2-5a8ffa7cf7a0)

A restart of PackageManager (from the menu) solved this temporarily issue and I could download & install GuiMods, and then finish RemoteGPIO install.

Except that, it worked like a charm from my perspective.

**Comment by kwindrem** [comment]
SetupHelper install does not defer GUI restarts since PackageManager restarts on package install and the state to prompt for the reboot. Not sure I can fix this but will look at it.

Conflict resolution does not download packages. I actually had not considered that but it is possible.
In any case, the lockup is related to a pending install of a package that is not in /data. Great catch, thanks!!!

Is downloading automatically a good thing or not? I will need to address the situation that the internet is not available !

Thanks for testing.

**Comment by Lucifer06** [comment]
I don't think the automatic download is required. the conflict message is clear, and user should be able to do the right action for resolving the issue.
But the automatic add of the right package is a good thing, especially when/if we reach a point where there dozens of packages listed in the repository.

**Comment by kwindrem** [comment]
Proceeding with conflict resolution that included installing a package that was not on the system locked up the process so I needed to address that.

In the end I got download + install working for conflict resolution. The prompt indicates what will happen before Proceed is pressed so I think it is OK that way.

If the conflicting package is not installed and the GitHub version isn't available for download, I note that to and hide the Proceed button.
